### ISSUE TITLE
Status of plugin?

### ISSUE THREAD

**Issue by AdrieanKhisbe** [asafdav/hapi-auth-extra#5]
Is plugin development still active, or had this been superseeded by the fork https://github.com/toymachiner62/hapi-authorization  ?
- If not, In what they differ
- If yes, would be good to add a pointer and deprecation notice :smiley:

**Comment by asafdav** [comment]
I wasn't aware of the forks, I'd love to see some pull requests and merge them in.

**Comment by AdrieanKhisbe** [comment]
Oh, I'm a bit surprised.
Then will be usefll to see if there a real divergence or now.

If hapi-authorization has changed of purpose.

@toymachiner62, would you mind joining the discussion ?  :smiley:

**Comment by toymachiner62** [comment]
Sure. When i was looking for authorization for my hapi app, I believe that hapi-auth-extra didn't provide all the functionality that we needed. So i created a fork and made hapi-authorization strictly for authorization and left authentication separately since we were using an internal authentication strategy.

I typically prefer single use libraries rather than a monolithic library so that's why hapi-authorizationis strictly authorization and does not do anything with authentication (especially since there already exist other authentication libs for hapi).

At least that's how i'm remembering the reasons why I forked hapi-authorization..

**Comment by AdrieanKhisbe** [comment]
I dig deeper into the two code source, running a diff.

The two main change are:
- removal of auth.js (delegated to other plugin, such as [hapi-auth-basic](https://github.com/hapijs/hapi-auth-basic) [cf example](https://github.com/toymachiner62/hapi-authorization#using-hapi-authorization-with-default-roles)
- disabling of system of _role hierarchy_ :  cf https://github.com/toymachiner62/hapi-authorization/blob/master/lib/roles.js#L22-L27

_@toymachiner62, tell me if you think I'm mistaken_

---

To me a convergence could be made, by reimplemnting a role hierarchy system. (which could really be useful),
and by renaming original @asafdav repositery.
`hapi-authorization` being more _name evocation_ that `auth-extra` 

@asafdav @toymachiner62, what do you think? :)

**Comment by asafdav** [comment]
Sounds good to me, what do you think @toymachiner62 ?

**Comment by toymachiner62** [comment]
Sorry, but what's the actual question you're asking of me?

**Comment by AdrieanKhisbe** [comment]
@toymachiner62. I was asking to confirm the two main changes I highlighted,

> removal of auth.js (delegated to other plugin, such as hapi-auth-basic cf example
>  disabling of system of role hierarchy : cf https://github.com/toymachiner62/hapi-authorization/blob/master/lib/roles.js#L22-L27

And if you were agreeing to the convergence suggestion :smiley:

> To me a convergence could be made, by reimplemnting a role hierarchy system. (which could really be useful),
> and by renaming original @asafdav repositery.
> hapi-authorization being more name evocation that auth-extra

**Comment by toymachiner62** [comment]
I don't think i removed the role hierarchy. https://github.com/toymachiner62/hapi-authorization/blob/master/lib/roles.js#L13-L20

So does your proposition of merging my code back mean that @asafdav is going to drop the authentication stuff from his code? 

I (and probably most other devs) prefer simple isolated packages rather than monolithic packages that perform multiple functions.

**Comment by AdrieanKhisbe** [comment]
> So does your proposition of merging my code back mean that @asafdav is going to drop the authentication stuff from his code?

Yep, that was implicitely implied.
Auth should be made with it's own module. https://github.com/dwyl/hapi-auth-jwt2 for instance

> I don't think i removed the role hierarchy. https://github.com/toymachiner62/hapi-authorization/blob/master/lib/roles.js#L13-L20

Indeed, I wasn't attentive enough. (hierarchy was more thinking me of a tree structure, that just a _clearance level_

**Comment by toymachiner62** [comment]
@asafdav Are you in agreement with the above https://github.com/asafdav/hapi-auth-extra/issues/5#issuecomment-165518086?

**Comment by asafdav** [comment]
@toymachiner62 Yes, we will have to create a migration plan for existing users but I agree with you, it would be better this way.

**Comment by AdrieanKhisbe** [comment]
:fireworks:  :)
